### ISSUE TITLE
Azure Storage Python Introduces Viral LGPL Dependencies

### ISSUE THREAD

**Issue by illfang** [Azure/azure-storage-python#514]
### Which service(blob, file, queue) does this issue concern?
blob

### Which version of the SDK was used? Please provide the output of `pip freeze`.
```
$ pipenv run pip install azure-storage-blob
...
$ pipenv run pip freeze
asn1crypto==0.24.0
azure-common==1.1.16
azure-nspkg==3.0.2
azure-storage-blob==1.3.1
azure-storage-common==1.3.0
azure-storage-nspkg==3.0.0
certifi==2018.10.15
cffi==1.11.5
chardet==3.0.4
cryptography==2.3.1
enum34==1.1.6
futures==3.2.0
idna==2.7
ipaddress==1.0.22
pycparser==2.19
python-dateutil==2.7.3
requests==2.19.1
six==1.11.0
urllib3==1.23
```

### What problem was encountered?
Installing azure-storage-blob library introduces a dependency to a LGPL library which infects any solution using the Azure Storage Library for Python.
Furthermore, there is no documentation whatsoever in the Azure Storage Library, that it relies on LGPL licensed software which is an important aspect which that should not be hidden.

Details:

https://github.com/Azure/azure-storage-python/blob/master/requirements.txt#L3
```
requests>=2.9.2
```

https://github.com/requests/requests/blob/master/setup.py#L51
```
'chardet>=3.0.2,<3.1.0',
```

To summarize this dependency chain:

`Azure Storage Python` -> `Requests >= 2.9.2` -> `chardet>=3.0.2,<3.1.0` 

Chardet License: https://github.com/chardet/chardet/blob/master/LICENSE
```
		  GNU LESSER GENERAL PUBLIC LICENSE
		       Version 2.1, February 1999
```

### Have you found a mitigation/solution?
Mocking away chardet installing azure blob without dependencies and manually add required dependencies by hand. Not a good solution.

Note: for table service, please post the issue here instead: https://github.com/Azure/azure-cosmosdb-python.

**Comment by zezha-msft** [comment]
Hi @illfang, thanks for reaching out!

I see that `chardet` is indeed an indirect dependency of our SDK, through the `requests` package, which is probably one of the most popular python libraries.

Please forgive my lack of knowledge about LGPL, but one [description of the license ](https://tldrlegal.com/license/gnu-lesser-general-public-license-v2.1-(lgpl-2.1))says:
> This license mainly applies to libraries. You may copy, distribute and modify the software provided that you state modifications and license them under LGPL-2.1. Anything statically linked to the library can only be redistributed under LGPL, but applications that use the library don't have to be.  You must allow reverse engineering of your application as necessary to debug and relink the library.

I do not see a problem since technically we are not distributing nor modifying the `chardet` package; the users of our SDK are installing that dependency directly from Pypi. In addition, if you are writing an application, this license also doesn't seem to affect you. 

This is really outside my area of expertise, so if you are sure that there is a problem, I can definitely escalate this issue to our legal department to double-check. Would you like me to do that?

**Comment by zezha-msft** [comment]
Hi @illfang, I understand your concern.

I will raise your issue with our legal team and get back to you as soon as possible.

**Comment by illfang** [comment]
Thank you for your investigation.
In our scenario we are thinking of a closed source client application distributed in a bundle to customers which allows the interaction with the Azure Blob Storage.

**Comment by illfang** [comment]
@zezha-msft I can understand your point of view on this issue. However, to be fair to the users of the storage library you should consider adding a NOTICE file pointing out, that this MIT licensed library will introduce LGPL licensed libraries as well.

**Comment by zezha-msft** [comment]
@illfang thanks for your understanding! And thanks for your advice. I'll sync up with our legal team to see if it's appropriate to put a notice file in the repo. 👍
